core/trino-spi/src/main/java/io/trino/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -181,7 +181,12 @@ public static BigInteger unscaledDecimalToBigInteger(Slice decimal)
         // convert to big-endian order
```

Comment:
The only reason I would like to stick with this approach is that 100% of sign-magnitude representations I know (namely 1 - floating point) uses 2 types of zeros.
I also implemented 128-bit arithmetic (partly) in C++ and I remember that trying to skip that negative zero cause a lot of unelegant code with branches and so on.

Reply:
> That's technically possible to have two different representation of zero, but it's definitely not desireable, because we will have toi remember about it in various places, including in the connectors.

Why is that a problem for connectors? Connectors can use `io.trino.spi.type.TypeOperators` for comparison (beware, MH are slow!)

Reply:
> Why is that a problem for connectors? 

A connector needs to deserialize data from stack representation (Slice) into something they operate on.
This happens in writes, but also in predicate pushdown.

The more flexibility runtime representation has, the more burden there is on a connector.

Reply:
> A connector needs to deserialize data from stack representation (Slice) into something they operate on.
This happens in writes, but also in predicate pushdown.

Why would connector bother with positive and negative zeros? It's engine responsibility to deal with it (either in comparison or packing)?

Reply:
Are we certain that no connectors use the existence of two zeros here?